src/sentry/utils/kafka.py

```diff
@@ -1,11 +1,14 @@
 import logging
 import signal
+from typing import Optional
 
 logger = logging.getLogger(__name__)
 
 
-def run_processor_with_signals(processor):
+def run_processor_with_signals(processor, consumer_name: Optional[str]):
```

Comment:
You'll have to add the delay before the subscription as well. 
Which means before building the `Processor`. As the subscription call is at the end of the initialization of the processor. 
The subscription call is the one that trigger the rebalance during start up. 
If you synchronize only the shutdown I don't think you will see any improvements as new consumers will be bootstrapped along the way triggering additional rebalances.

Reply:
yeah the delay in subscription would be introduced here https://github.com/getsentry/sentry/pull/62018/files#diff-b3d2a21f7b228f7f3adc500497948ca6a7288d43712bde3b90fe67ee723c9d14R612-R615. the subscribe happens in the stream processor constructor. by having the delay here, we essentially delay the whole app from instantiating the stream processor

Reply:
Ok, good

Reply:
https://github.com/getsentry/sentry/pull/62018#issuecomment-1863327946